app/api/schema/access_codes.py

```diff
@@ -27,15 +27,19 @@ class Meta:
 
     @validates_schema(pass_original=True)
     def validate_date(self, data, original_data):
+        ends_at = data.get('valid_till', None)
         if 'id' in original_data['data']:
             access_code = AccessCode.query.filter_by(id=original_data['data']['id']).one()
 
             if 'valid_from' not in data:
                 data['valid_from'] = access_code.valid_from
 
-            data['valid_till'] = access_code.valid_expire_time
+            if 'valid_till' not in data:
+                data['valid_till'] = discount_code.valid_till
 
-        if data['valid_till'] and data['valid_from'] > data['valid_till']:
+            ends_at = data.get('valid_till') or discount_code.valid_till
+
+        if ends_at and data['valid_from'] > ends_at:
```

Comment:
```suggestion
        if data['valid_from'] > ends_at:
```

Reply:
why removing `ends_at`. while creating the discount code, and choose when sales ends. than `ends_at=data.get('valid_till', None)` will return None and data['valid_from'] > ends_at: is giving error ``TypeError: '>' not supported between instances of 'datetime.datetime' and 'NoneType'``